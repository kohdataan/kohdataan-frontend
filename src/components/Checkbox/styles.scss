@import '../../styles/colors.scss';

/* Customize the label (the container) */
.container {
  display: block;
  position: relative;
  padding-left: 33px;
  margin-bottom: 23px;
  cursor: pointer;
  font-size: 18px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

/* Hide the browser's default checkbox */
.container input {
  position: absolute;
  opacity: 0;
  cursor: pointer;
  height: 0;
  width: 0;
}

/* Create a custom checkbox */
.checkmark {
  position: absolute;
  top: 0;
  left: 0;
  height: 24px;
  width: 24px;
  background-color: $white;
  border-radius: 2px;
  border: solid 2px $orange;
}

/* Create the checkmark/indicator (hidden when not checked) */
.checkmark:after {
  content: "";
  position: absolute;
  display: none;
}

/* Show the checkmark when checked */
.container input:checked ~ .checkmark:after {
  display: block;
}

/* Style the checkmark/indicator */
.container .checkmark:after {
  left: 7px;
  top: 3px;
  width: 5px;
  height: 10px;
  border: solid $black;
  border-width: 0 3px 3px 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}
