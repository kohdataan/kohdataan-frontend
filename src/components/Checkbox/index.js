import React, { memo } from 'react'
import PropTypes from 'prop-types'
import './styles.scss'

const Checkbox = (props) => {
  const {
    name,
    labelClassName,
    inputClassName,
    value,
    onChange,
    label,
    checked,
  } = props
  return (
    <label htmlFor={name} className={`checkbox ${labelClassName}`}>
      {label}
      <input
        type="checkbox"
        name={name}
        id={name}
        value={value}
        checked={checked}
        className={inputClassName}
        onChange={onChange}
      />
      <span className="checkmark" />
    </label>
  )
}

Checkbox.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.string,
  checked: PropTypes.bool,
  inputClassName: PropTypes.string,
  labelClassName: PropTypes.string,
  onChange: PropTypes.func.isRequired,
}

Checkbox.defaultProps = {
  value: '',
  checked: false,
  inputClassName: '',
  labelClassName: '',
}

export default memo(Checkbox)
