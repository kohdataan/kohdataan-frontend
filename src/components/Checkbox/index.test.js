import React from 'react'
import { create } from 'react-test-renderer'
import Checkbox from './index'

describe('Checkbox', () => {
  test('it matches the snapshot', () => {
    const component = create(
      <Checkbox label="test" name="test" onChange={() => {}} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
