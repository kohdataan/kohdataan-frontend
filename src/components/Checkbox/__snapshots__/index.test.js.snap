// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox it matches the snapshot 1`] = `
<label
  className="checkbox "
  htmlFor="test"
>
  test
  <input
    checked={false}
    className=""
    id="test"
    name="test"
    onChange={[Function]}
    type="checkbox"
    value=""
  />
  <span
    className="checkmark"
  />
</label>
`;
